app/common/elasticsearchclient.py

```diff
@@ -1484,6 +1524,44 @@ def _free_text_query(self, searchphrase, doc_types, params):
 
                                    )
 
+    def _best_hit_query(self, doc_types, params):
+        '''
+           If  'fields' parameter is passed, only these fields would be returned
+           and 'highlights' would be added only if it is of the fields parameters.
+           If there is not a 'fields' parameter, then fields are included by default
+
+        '''
+
+
+        the_head = {'index':self._index_search, 'type':doc_types}
+        multi_body = [];
+        the_highlight = self._get_free_text_highlight()
+        source = SourceDataStructureOptions.getSource(params.datastructure, params)
+
+        if 'include' in source:
+            params.requested_fields = source['include']
+            if 'highlight' not in params.requested_fields:
+                the_highlight = None
+
+        for searchphrase in params.q:
+            the_body = {'query': self._get_free_text_query(searchphrase),
+                        'size': params.size,
+                        'from': params.start_from,
+                        '_source': source,
+                        "explain": current_app.config['DEBUG']
+                        }
+            if the_highlight is not None:
+                the_body['highlight'] = the_highlight
+            multi_body.append(the_head)
+            multi_body.append(the_body)
+        
+#         request = ''
+#         for each in multi_body:
+#             request += '%s \n' %json.dumps(each)    
+
+        return self._cached_search(body=multi_body,
```

Comment:
I am a bit confused here : let me see if i understand - you want separate resulting data into 3 different lists - one would be list that has all the responses that are fuzzy, one that has exact and one that did not find anything.